src/timeMachine/selectors/index.ts

```diff
@@ -147,6 +148,16 @@ export const getXColumnSelection = (state: AppState): string => {
   return defaultXColumn(table, preferredXColumnKey)
 }
 
+export const getXColumnTimeDomainBounds = (
+  state: AppState
+): [number, number] => {
+  const {table} = getVisTable(state)
+
+  const times = table.getColumn('_time')
+
+  return times ? [times[0] as number, times[times.length - 1] as number] : null
```

Comment:
this is a bit unsafe and inferring. it infers that if the `times` is truthy, i'll be an array. That's a bit dangerous.

I think it would improve readability and make the code more explicit to expand this out into one or more guard statements like:

```typescript
if (!Array.isArray(times)) {
  return null
}

if (times.length < 2) {
  return null
}

return [times[0] as number, times[times.length - 1] as number]
```